### DISCUSSION TITLE
Why Code Server starts but I failed to connect to it?

### DISCUSSION THREAD

**Discussion by kanadeCN** [coder/code-server#6756] [Q&A]
I’m trying to run codeserver on my iPad  through [UTM ](https://github.com/utmapp/UTM) to code locally. The codeserver had been installed to a Debian12(arm64) virtual machine. It seemed to be successful but I failed to connect it when opening http://127.0.0.1:1314/
![778D59F1-07AC-4B7E-822E-4A0DE62DBA57](https://github.com/coder/code-server/assets/63999920/be445b2c-7635-4ed7-a53f-d688e8cefcf7)
 
![55217A74-480B-42B0-AC46-0748266942FF](https://github.com/coder/code-server/assets/63999920/ec10b277-bc24-4379-8d3c-3fd97f0a94ed)

![4E47757B-E128-44EE-9DB1-E0F412ECAAE9](https://github.com/coder/code-server/assets/63999920/84c72426-45a9-4ee0-bebb-38fa14ff85a2)

**Comment by MC-Spraint** [comment]
I also have the same issue, I have deployed code-server on AWS ec2 instance and currently is running on port 8080.
But when I try to access it publicly via the url(http:my_ip:8080), I can't access it. Also to mention when I run other servers on port 8080 I can publicly access it. Only in this case its inaccessible. If you find a way to address this issue, please let me know. I'll be waiting

**Reply by code-asher** [reply]
Just to be sure, did you also bind code-server to `0.0.0.0`?  If it is on `127.0.0.1` it will not be accessible outside the machine with the IP.

**Reply by code-asher** [reply]
I see there is another discussion for this, I will copy my comment so let us continue the conversation there!

**Comment by code-asher** [comment]
I have no experience with UTM so I can only guess, but it looks like you might need to forward the port out of the VM?

https://docs.getutm.app/settings-qemu/devices/network/port-forwarding/

If that does not help, you might have to talk to the UTM folks.  You could try reproducing with another program just to rule out issues with code-server, for example `python3 -m http.server`.
